lib/loader/loader.js

```diff
@@ -130,29 +139,40 @@ var loader, define, requireModule, require, requirejs;
   };
 
   Module.prototype.unsee = function() {
-    this.finalized = false;
-    this._foundDeps = false;
-    this.isPending = false;
+    this.state = 'new';
     this.module = { exports: {} };
   };
 
   Module.prototype.reify = function() {
+    if (this.state === 'reified') { return; }
+    this.state = 'reifying';
+    try {
+      this.reified = this._reify();
+      this.state = 'reified';
+    } finally {
+      if (this.state === 'reifying') {
+        this.state = 'errored';
+      }
+    }
+  };
+
+  Module.prototype._reify = function() {
     stats.reify++;
-    var reified = this.reified;
+    var reified = this.reified.slice();
```

Comment:
Yes. We don't want to end up with a half-reified array if one of these dependencies throws.

My goal here is to make sure that when you re-enter the code path, you will get back to the exception and hit it again. If we had already reified the prefix of the array, you would get an unrelated exception instead.